tests/foreman/api/test_remoteexecution.py

```diff
@@ -72,3 +75,137 @@ def test_positive_run_capsule_upgrade_playbook(module_capsule_configured, target
     ).refresh()
     feature_set = {feat['name'] for feat in result['features']}
     assert {'Ansible', 'Dynflow', 'Script', 'Pulpcore', 'Logs'}.issubset(feature_set)
+
+
+@pytest.mark.tier3
+@pytest.mark.no_containers
+@pytest.mark.rhel_ver_match('[^6].*')
+def test_negative_time_to_pickup(
+    module_org,
+    module_target_sat,
+    smart_proxy_location,
+    module_ak_with_cv,
+    module_capsule_configured_mqtt,
+    rhel_contenthost,
+):
+    """Time to pickup setting is honored for host registered to mqtt
+
+    :id: a082f599-fbf7-4779-aa18-5139e2bce774
+
+    :expectedresults: Time to pickup aborts the job if mqtt client doesn't
+        respond in time
+
+    :CaseImportance: High
+
+    :bz: 2158738, 2118651
+
+    :parametrized: yes
+    """
+    client_repo = ohsnap.dogfood_repository(
+        settings.repos.ohsnap_repo_host,
+        product='client',
+        repo='client',
+        release='Client',
+        os_release=rhel_contenthost.os_version.major,
+    )
+    # Update module_capsule_configured_mqtt to include module_org/smart_proxy_location
+    module_target_sat.cli.Capsule.update(
+        {
+            'name': module_capsule_configured_mqtt.hostname,
+            'organization-ids': module_org.id,
+            'location-ids': smart_proxy_location.id,
+        }
+    )
+    # register host with pull provider rex
+    result = rhel_contenthost.register(
+        module_org,
+        smart_proxy_location,
+        module_ak_with_cv.name,
+        target=module_capsule_configured_mqtt,
+        satellite=module_target_sat,
+        setup_remote_execution_pull=True,
+        repo=client_repo.baseurl,
+    )
+    template_id = (
+        module_target_sat.api.JobTemplate()
+        .search(query={'search': 'name="Run Command - Script Default"'})[0]
+        .id
+    )
+    assert result.status == 0, f'Failed to register host: {result.stderr}'
+    # check mqtt client is running
+    result = rhel_contenthost.execute('systemctl status yggdrasild')
+    assert result.status == 0, f'Failed to start yggdrasil on client: {result.stderr}'
+    # check that longrunning command is not affected by time_to_pickup BZ#2158738
+    job = module_target_sat.api.JobInvocation().run(
+        synchronous=False,
+        data={
+            'job_template_id': template_id,
+            'organization': module_org.name,
+            'location': smart_proxy_location.name,
+            'inputs': {
+                'command': 'echo start; sleep 10; echo done',
+            },
+            'targeting_type': 'static_query',
+            'search_query': f'name = {rhel_contenthost.hostname}',
+            'time_to_pickup': '5',
+        },
+    )
+    wait_for_tasks(f'resource_type = JobInvocation and resource_id = {job["id"]}')
+    result = module_target_sat.api.JobInvocation(id=job['id']).read()
+    assert result.succeeded == 1
```

Comment:
I guess the `job` variable will have  `result`  to assert `success`
https://github.com/SatelliteQE/robottelo/blob/68aba5f021b6c8dbe16f9746f2accd27457502a7/tests/foreman/api/test_provisioning.py#L137
```suggestion
    assert job['result'] == 'success'
```

Reply:
at the time the job variable is populated, it doesn't yet have result